packages/SwingSet/src/vats/comms/clist-inbound.js

```diff
@@ -29,46 +19,43 @@ export function makeInbound(state, stateKit) {
 
   function retireRemotePromiseID(remoteID, rpid) {
     insistRemoteType('promise', rpid);
-    const remote = getRemote(state, remoteID);
-    const lpid = remote.fromRemote.get(rpid);
+    const remote = state.getRemote(remoteID);
+    const lpid = remote.mapFromRemote(rpid);
     assert(lpid, X`unknown remote ${remoteID} promise ${rpid}`);
-    const p = state.promiseTable.get(lpid);
+    const p = state.getPromise(lpid);
     assert(
       !p.subscribers || p.subscribers.indexOf(remoteID) === -1,
       X`attempt to retire remote ${remoteID} subscribed promise ${rpid}`,
     );
-    remote.fromRemote.delete(rpid);
-    remote.toRemote.delete(lpid);
+    remote.deleteRemoteMapping(rpid, lpid);
     cdebug(`comms delete mapping r<->k ${remoteID} {rpid}<=>${lpid}`);
   }
 
   function beginRemotePromiseIDRetirement(remoteID, rpid) {
     insistRemoteType('promise', rpid);
-    const remote = getRemote(state, remoteID);
-    const lpid = remote.fromRemote.get(flipRemoteSlot(rpid));
-    remote.toRemote.delete(lpid);
-    remote.retirementQueue.push([remote.nextSendSeqNum, rpid]);
+    const remote = state.getRemote(remoteID);
+    const lpid = remote.mapFromRemote(flipRemoteSlot(rpid));
+    remote.deleteToRemoteMapping(lpid);
+    remote.enqueueRetirement([remote.nextSendSeqNum(), rpid]);
```

Comment:
This feels like an `enqueueRetirement(rpid)` should look up the next seqnum internally.